### PR TITLE
Add Config Merger for k8s in Canary

### PR DESCRIPTION
**Pull request by chases2**
Also allows config_merger to read from a URL with a special flag.

Do not merge until https://github.com/kubernetes/test-infra/pull/20986 merges, or the URL won't work.

### CODE REVIEW

File: cluster/canary/config_merger.yaml
Comment (chases2) [inline cluster/canary/config_merger.yaml:30]: Note: I moved this, so the URL will be different. Will fix after other PR merges.

**Review by michelle192837 — APPROVED** [review]
/hold

Unhold at will!

**Review by e-blackwelder — APPROVED** [review]

**Pull request merged**
